### PR TITLE
Remove the unnecessary cast for main_t, NULL and argv

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
I have try my best to fix the style issue, so please ignore the rest style warning.

## Impact
Minor, code refactor

## Testing
Pass CI

### CODE REVIEW

File: graphics/nxwidgets/UnitTests/CButton/cbuttontest.cxx [hunk 1]
```diff
@@ -154,7 +154,7 @@ bool CButtonTest::createWindow(void)
 {
   // Initialize the widget control using the default style
 
-  m_widgetControl = new CWidgetControl((CWidgetStyle *)NULL);
+  m_widgetControl = new CWidgetControlNULL);
```
Comment (pkarashchenko) [inline graphics/nxwidgets/UnitTests/CButton/cbuttontest.cxx:157]:
```suggestion
  m_widgetControl = new CWidgetControl(NULL);
```

File: graphics/nxwidgets/UnitTests/CButton/cbuttontest.cxx [hunk 2]
```diff
@@ -64,9 +64,9 @@
 
 CButtonTest::CButtonTest()
 {
-  m_bgWindow = (CBgWindow *)NULL;
-  m_nxFont   = (CNxFont *)NULL;
-  m_text     = (CNxString *)NULL;
+  m_bgWindow = NULL;
+  m_nxFont   = NULL;
+  m_text     = NULL;
```
Comment (pkarashchenko) [inline graphics/nxwidgets/UnitTests/CButton/cbuttontest.cxx:69]: just for my understanding. a generic question not related to this PR. shouldn't cxx code use `nullptr`?

File: graphics/nxwidgets/UnitTests/CButtonArray/cbuttonarraytest.cxx [hunk 3]
```diff
@@ -134,7 +134,7 @@ bool CButtonArrayTest::createWindow(void)
 {
   // Initialize the widget control using the default style
 
-  m_widgetControl = new CWidgetControl((CWidgetStyle *)NULL);
+  m_widgetControl = new CWidgetControlNULL);
```
Comment (pkarashchenko) [inline graphics/nxwidgets/UnitTests/CButtonArray/cbuttonarraytest.cxx:137]:
```suggestion
  m_widgetControl = new CWidgetControl(NULL);
```

File: graphics/nxwidgets/UnitTests/CCheckBox/ccheckboxtest.cxx [hunk 4]
```diff
@@ -149,7 +149,7 @@ bool CCheckBoxTest::createWindow(void)
 {
   // Initialize the widget control using the default style
 
-  m_widgetControl = new CWidgetControl((CWidgetStyle *)NULL);
+  m_widgetControl = new CWidgetControlNULL);
```
Comment (pkarashchenko) [inline graphics/nxwidgets/UnitTests/CCheckBox/ccheckboxtest.cxx:152]:
```suggestion
  m_widgetControl = new CWidgetControl(NULL);
```

File: graphics/nxwidgets/UnitTests/CGlyphButton/cglyphbuttontest.cxx [hunk 5]
```diff
@@ -136,7 +136,7 @@ bool CGlyphButtonTest::createWindow(void)
 {
   // Initialize the widget control using the default style
 
-  m_widgetControl = new CWidgetControl((CWidgetStyle *)NULL);
+  m_widgetControl = new CWidgetControlNULL);
```
Comment (pkarashchenko) [inline graphics/nxwidgets/UnitTests/CGlyphButton/cglyphbuttontest.cxx:139]:
```suggestion
  m_widgetControl = new CWidgetControl(NULL);
```

File: graphics/pdcurs34/pdcurses/pdc_terminfo.c [hunk 6]
```diff
@@ -244,15 +244,15 @@ char *tigetstr(const char *capname)
 {
   PDC_LOG(("tigetstr() - called: capname %s\n", capname));
 
-  return (char *)(-1);
+  return (-1);
```
Comment (pkarashchenko) [inline graphics/pdcurs34/pdcurses/pdc_terminfo.c:247]: why cast is removed here? Isn't compiler warning produced?

File: netutils/thttpd/libhttpd.c [hunk 7]
```diff
@@ -1637,7 +1671,8 @@ static void ls_child(int argc, char **argv)
             {
               oldmax    = maxnames;
               maxnames *= 2;
-              names     = RENEW(names, char, oldmax*(PATH_MAX + 1), maxnames*(PATH_MAX + 1));
+              names     = RENEW(names, char, oldmax*(PATH_MAX + 1),
+                                maxnames*(PATH_MAX + 1));
```
Comment (pkarashchenko) [inline netutils/thttpd/libhttpd.c:1675]:
```suggestion
              names     = RENEW(names, char, oldmax * (PATH_MAX + 1),
                                maxnames * (PATH_MAX + 1));
```

File: graphics/nxwidgets/UnitTests/CButton/cbuttontest.cxx (hunk 2 above)
Comment (xiaoxiang781216) [reply graphics/nxwidgets/UnitTests/CButton/cbuttontest.cxx:69]: nullptr doesn't exist before c++11, NULL will be defined to nullptr after c++11.

File: examples/poll/select_listener.c [hunk 8]
```diff
@@ -77,18 +77,18 @@ void *select_listener(pthread_addr_t pvarg)
...
   printf("select_listener: Opening %s for non-blocking read\n", FIFO_PATH2);
-  fd = open(FIFO_PATH2, O_RDONLY|O_NONBLOCK);
+  fd = open(FIFO_PATH2, O_RDONLY | O_NONBLOCK);
   if (fd < 0)
     {
       printf("select_listener: ERROR Failed to open FIFO %s: %d\n",
              FIFO_PATH2, errno);
       close(fd);
-      return (void*)-1;
+      return (void *)-1;
```
Comment (pkarashchenko) [inline examples/poll/select_listener.c:86]:
```suggestion
      return (void *)(uintptr_t)-1;
```

**Review by pkarashchenko — APPROVED** [review]

File: examples/poll/select_listener.c (hunk 8 above)
Comment (xiaoxiang781216) [reply examples/poll/select_listener.c]: Done.

**Review by pkarashchenko — APPROVED** [review]

**Review by pkarashchenko — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
@pkarashchenko could you merge this patch?

**Pull request merged**